src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -486,23 +559,154 @@ InsertValidationResponse insertBufferedRecords(StreamingBuffer streamingBufferTo
     return response;
   }
 
-  /**
-   * Uses {@link Fallback} API to reopen the channel if insertRows throws {@link SFException}.
-   *
-   * <p>We have deliberately not performed retries on insertRows because it might slow down overall
-   * ingestion and introduce lags in committing offsets to Kafka.
-   *
-   * <p>Note that insertRows API does perform channel validation which might throw SFException if
-   * channel is invalidated.
-   *
-   * <p>It can also send errors {@link
-   * net.snowflake.ingest.streaming.InsertValidationResponse.InsertError} in form of response inside
-   * {@link InsertValidationResponse}
-   *
-   * @param buffer buffer to insert into snowflake
-   * @return InsertValidationResponse object sent from insertRows API.
-   */
-  private InsertValidationResponse insertRowsWithFallback(StreamingBuffer buffer) {
+  public class InsertRowsWithRetryResponse {
```

Comment:
java docs for this class.

Reply:
Can you rename this to InsertRowsReponse and instead put retry count inside this class. Class name should be generic and more details should be part of the instance variables.